### PR TITLE
TDD knight valid_move?

### PR DESCRIPTION
**Pull request by icrustandi**
### What does this PR do?
Adds a valid_move? method for knight. 

### Where should I start?
app/models/knight.rb
spec/models/knight_spec.rb

### How do I manually test this?
While in the pancakes project folder, run the following in console:
rspec 'spec/models/knight_spec.rb'

### Additional Comments
This valid_move method uses a correct_move? helper method to determine if the piece has moved a distance of 3 squares.

### CODE REVIEW

File: spec/factories/knights.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FactoryGirl.define do
```
Comment (danielpowell4) [inline spec/factories/knights.rb]: Is this being used at all?

Seems like no as you're creating Pieces in the test rather than here with traits--

Comment (icrustandi) [inline spec/factories/knights.rb]: Yeah, you're right - I'm not using that factory at all in this commit. I used 'rails generate rspec:model knight' to generate this test, so rspec creates that factory automatically. We might need it in the future for other tests, so I left it there for now.

**Comment by danielpowell4** [comment]
Like the tests!

I left a comment on the factories, which seemed unused.

Is `valid_move?` checking whether a move is legal and `correct_move?` checking that it's the right length?

Do you like these names? What do you think about `legal_move?` and `proper_length?`

**Comment by icrustandi** [comment]
I like your suggestions Dan. I'll make those name changes and push it back up to the PR

**Pull request merged**